axiom/nr_segment.c

```diff
@@ -1170,23 +1170,59 @@ void nr_segment_record_exception(nr_segment_t* segment,
   nr_segment_set_error(segment, error_message, error_class);
 }
 
-void nr_segment_set_error(nr_segment_t* segment,
-                          const char* error_message,
-                          const char* error_class) {
-  if ((NULL == segment) || (NULL == error_message && NULL == error_class)) {
-    return;
-  }
-
+static void nr_segment_set_error_helper(nr_segment_t* segment,
+                                        const char* error_message,
+                                        const char* error_class,
+                                        const char* error_file,
+                                        int error_line,
+                                        char* error_context,
+                                        int error_no) {
   if (NULL == segment->error) {
     segment->error = nr_zalloc(sizeof(nr_segment_error_t));
   }
 
   nr_free(segment->error->error_message);
   nr_free(segment->error->error_class);
+  if (NULL != error_file && NULL != error_context) {
```

Comment:
This code seems like it could leak.  If the caller doesn't define these values we don't free the current values.  Below the current value is set to `NULL` if the caller doesn't define a value, thus losing a reference to the memory?